lxd-agent/response.go

```diff
@@ -12,19 +13,38 @@ type devLXDResponse struct {
 	content any
 	code    int
 	ctype   string
+	etag    string
 	hook    func(w http.ResponseWriter) error
 }
 
 // Render renders a devLXD response.
 func (r *devLXDResponse) Render(w http.ResponseWriter, req *http.Request) error {
 	var err error
 
-	// Write response.
+	// Handle hooks first, if defined.
 	if r.hook != nil {
-		err = r.hook(w)
-	} else if r.code != http.StatusOK {
+		return r.hook(w)
+	}
+
+	// Handle error responses.
+	if r.code != http.StatusOK {
 		http.Error(w, fmt.Sprint(r.content), r.code)
-	} else if r.ctype == "json" {
+		return nil
+	}
+
+	// Set ETag if provided.
+	if r.etag != "" {
+		// Remove quotes from ETag if present.
```

Comment:
Still not clear on this quoting business. Seems like we should be breaking this out into separate function(s) to de-dupe this quoting logic as ive seen it in a few places now.